An implicit cast from `const char *` to `char *` is not possible without
adding `-fpermissive`. But let's rather make the cast explicit here.
--- a/gdb/completer.c
+++ b/gdb/completer.c
@@ -2011,7 +2011,7 @@
       rl_basic_quote_characters = NULL;
     }
 
-  return rl_completer_word_break_characters;
+  return (char *)rl_completer_word_break_characters;
 }
 
 char *
